Magento EQP Documentation update. (#8112)
magento/devdocs: src/marketplace/sellers/code-sniffer.md

---
group: marketplace-sellers
title: Code Sniffer
---

## Overview

> Code Sniffer is a static test that allows the detection of violations of the [Magento Coding Standard](https://github.com/magento/magento-coding-standard/) and prevents common issues by static code analysis.

Reviewer: ```suggestion
Code Sniffer is a static test that detects violations of the [Magento Coding Standard](https://github.com/magento/magento-coding-standard/) and prevents common issues by static code analysis.
```

## What is being checked

Code Sniffer validates that the implementation of the submitted extension adheres to the [Magento Coding Standard](https://github.com/magento/magento-coding-standard/).

## Tools and environments used

The Magento EQP Code Sniffer is based on the [PHP_CodeSniffer](https://github.com/squizlabs/PHP_CodeSniffer) and uses [Magento Coding Standard](https://github.com/magento/magento-coding-standard/) as a ruleset.

You can complete Code Sniffer testing in a development environment by using Composer to install PHPCS and the Magento Coding Standard as a global dependency for a particular project.

You can test an extension to detect violations of the [Magento Coding Standard](https://github.com/magento/magento-coding-standard/) using the following [PHPCS CLI command](https://github.com/squizlabs/PHP_CodeSniffer/blob/master/bin/phpcs):

```bash
phpcs --standard=Magento2 --extensions=php,phtml --error-severity=10 --ignore-annotations --report=json --report-file=report.json <path-to-extension>
```
If PHPCS finds any errors, the extension in `<path-to-extension>`  is rejected.

## Reading the error report

> All detected errors are self-explanatory and have references to the locations in the code where they were detected.

Reviewer: ```suggestion
All detected errors include a description of the problem with references to the locations in the code where the errors were detected.
```
